requirements/requirements.txt

```diff
@@ -1,110 +1,165 @@
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    make requirements
+#
+-e git+git://github.com/dimagi/architect@v0.5.7a2#egg=architect
+alembic==0.8.6
+amqp==1.4.9
+anyjson==0.3.3            # via kombu
+appdirs==1.4.3            # via zeep
+asn1crypto==0.24.0        # via cryptography
 attrs==18.1.0
-# needed to build gevent from source
-cython==0.27.3
-restkit==4.2.2
-iso8601==0.1.11
-jsonobject==0.9.1
-jsonobject-couchdbkit==0.9.1
+babel==2.6.0              # via django-phonenumber-field
+billiard==3.3.0.23        # via celery
+boto3==1.4.4
+botocore==1.5.95          # via boto3, s3transfer
+cached-property==1.4.3    # via zeep
 celery==3.1.18
-# required by django-digest
+cffi==1.11.5              # via cryptography, csiphash
+chardet==3.0.4            # via ghdiff
+cloudant==2.7.0           # via jsonobject-couchdbkit
+concurrentloghandler==0.9.1
+contextlib2==0.5.4
+cryptography==2.3         # via pyopenssl
+csiphash==0.0.5
+csv342==1.0.0
+cython==0.27.3
+datadog==0.15.0
 decorator==4.0.11
-django==1.11.14
-django-braces==1.8.1
-django-celery==3.2.1
-django-crispy-forms==1.7.0
-jsonfield==1.0.3
-dropbox==7.3.1
 defusedxml==0.4.1
 diff-match-patch==20120106
 dimagi-memoized==1.0.1
+django-angular==0.7.16
+django-appconf==1.0.2     # via django-compressor, django-statici18n
+django-braces==1.8.1
+django-bulk-update==1.1.10
+django-celery==3.2.1
+django-compressor==2.1
+django-countries==4.6
+django-crispy-forms==1.7.0
+django-cte==1.1.2
+django-formtools==2.0
+django-otp==0.3.13
+django-phonenumber-field==1.3.0  # via django-two-factor-auth
+django-prbac==0.0.5
+django-ranged-response==0.2.0  # via django-simple-captcha
+django-redis-sessions==0.5.6
+django-redis==4.2
+django-simple-captcha==0.5.7
+django-statici18n==1.3.0
 django-tastypie==0.14.0
+django-transfer==0.4
+django-two-factor-auth==1.6.2
+django-user-agents==0.3.2
+django-websocket-redis==0.5.1
+django==1.11.14
 djangorestframework==3.5.4
+dnspython==1.15.0
+docutils==0.14            # via botocore
+dropbox==7.3.1
 elasticsearch==1.9.0
+email_validator==1.0.2
+enum34==1.1.6             # via cryptography
+et-xmlfile==1.0.1         # via openpyxl
+ethiopian-date-converter==0.1.4
 eulxml==1.1.3
+faker==0.8.15
+funcsigs==1.0.2           # via mock
+futures==3.2.0            # via s3transfer, tinys3
```

Comment:
Including ```futures``` causes the python 3 docker build to fail, because the library is not provided for python 3.  In fact, it's not specified as a dependency for these libraries in python 3: [s3transfer](https://github.com/boto/s3transfer/blob/0e692c364fc0074bdee9c9836605927b3e044369/setup.py#L18-L21) and [tinys3](https://github.com/smore-inc/tinys3/blob/7fd6aa53bdf245a992af5e3c61a37fb5bf2acc62/setup.py#L10-L11).

I imagine we'd want a separate version of the requirements files to be generated for python 3.  Do you know if ```pip-compile``` provides support for this?  Otherwise we'd need a post-processing step to remove libraries not included in python 3 (I'm only thinking of ```futures``` at the moment).

Reply:
That's a very interesting question. I think what it would take is running `pip-compile` once in a python2 venv, and once in a python3 venv. Would you be able to do a little online research on what other people do for this? (If 30-60 minutes doesn't lead to a useful direction, let's set a time to investigate and/or discuss next steps together.)